<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="rviz" default="true" />
  
  
  <arg name="configfile" default="$(find covgrid_slam)/config/configpcd.xml" />
  <arg name="logdir" default="--log_dir=/home/jkj/catkin_ws/log" />
  <arg name="glogparam" default="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=5   $(arg logdir)" />

  <node pkg="covgrid_slam" type="masternode" name="masternode" args="$(arg glogparam)" output="screen">
  	<param name="config_file" value="$(arg configfile)"/>
  </node>

  
  <node pkg="covgrid_slam" type="lidarodometrymain" name="lidarodometrymain" args="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=0   $(arg logdir)" output="screen">
  <param name="configuration_directory" value="$(find covgrid_slam)/config" />
  <param name="configuration_basename" value="localmap.lua" />
  <param name="withoutimu" value="false"/>
  </node>

  <node pkg="covgrid_slam" type="transformmain" name="transformmain" args="$(arg glogparam)" output="screen">
  </node>

  <node pkg="covgrid_slam" type="getinsdata" name="getinsdata" args="$(arg glogparam)" output="screen">
  </node>
  
  <node pkg="covgrid_slam" type="getvelodynedata" name="getvelodynedata" args="$(arg glogparam)" output="screen">
  </node>

  
  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find covgrid_slam)/rviz_cfg/covgrid_slam_new.rviz" />
  </group>

</launch>
